viz/cmd/authz.go

```diff
@@ -89,62 +91,55 @@ func NewCmdAuthz() *cobra.Command {
 				os.Exit(1)
 			}
 			for _, server := range servers {
-				sazs, err := k8s.ServerAuthorizationsForServer(cmd.Context(), k8sAPI, options.namespace, server)
+				requestParams := util.StatsSummaryRequestParams{
+					StatsBaseRequestParams: util.StatsBaseRequestParams{
+						TimeWindow:    options.timeWindow,
+						ResourceName:  server,
+						ResourceType:  k8s.Server,
+						Namespace:     options.namespace,
+						AllNamespaces: false,
+					},
+				}
+
+				req, err := util.BuildStatSummaryRequest(requestParams)
+				if err != nil {
+					return err
+				}
+				resp, err := requestStatsFromAPI(client, req)
 				if err != nil {
 					fmt.Fprint(os.Stderr, err.Error())
 					os.Exit(1)
 				}
-				for _, saz := range sazs {
-					requestParams := util.StatsSummaryRequestParams{
-						StatsBaseRequestParams: util.StatsBaseRequestParams{
-							TimeWindow:    options.timeWindow,
-							ResourceName:  saz,
-							ResourceType:  k8s.ServerAuthorization,
-							Namespace:     options.namespace,
-							AllNamespaces: false,
-						},
-						ToNamespace: options.namespace,
-					}
-					requestParams.ToName = server
-					requestParams.ToType = k8s.Server
 
-					req, err := util.BuildStatSummaryRequest(requestParams)
-					if err != nil {
-						return err
+				for _, row := range respToRows(resp) {
+					var authzp, saz, route string
+					if row.GetSrvStats().GetAuthz() != nil {
+						if row.GetSrvStats().GetAuthz().Type == "authorizationpolicy" {
+							authzp = row.GetSrvStats().GetAuthz().Name
+						} else if row.GetSrvStats().GetAuthz().Type == "serverauthorization" {
+							saz = row.GetSrvStats().GetAuthz().Name
+						}
 					}
-					resp, err := requestStatsFromAPI(client, req)
-					if err != nil {
-						fmt.Fprint(os.Stderr, err.Error())
-						os.Exit(1)
+					if row.GetSrvStats().GetRoute() != nil {
+						route = row.GetSrvStats().GetRoute().Name
 					}
-
-					for _, row := range respToRows(resp) {
-						if row.Stats == nil {
-							rows = append(rows, table.Row{
-								server,
-								saz,
-								"-",
-								"-",
-								"-",
-								"-",
-								"-",
-							})
```

Comment:
it looks like we are now skipping rows with no stats? do we still want to display them in the list even if we don't have stats for them?

Reply:
potentially, but it's not obvious exactly which rows to show.  The command works by iterating over Servers and doing a metrics query for each one.  If the control plane has returned nil stats, it means that no traffic has been measured for that Server, which means that we don't have any authorization labels either.  I opted to omit this entirely since this command is supposed to show authorizations.  An alternative would be to list a row for every authorization which targets the Server with blank stats for each one, but this seems excessive to me.